### PR TITLE
Publish packages in CI and update to crossplane-contrib orgs

### PR DESCRIPTION
**Pull request by jastang**
Signed-off-by: Jason Tang <jason@upbound.io>

### Description of your changes
Update the CI to publish to `crossplane-contrib` OCI repos in both DockerHub and Upbound registries.

xref https://github.com/crossplane/crossplane/issues/3350

Also updates `crossplane.yaml` to use the updated metadata annotations per the xpkg spec so the package will render nicely in the Upbound Marketplace.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

*Publish*
```
➜  provider-argocd git:(jastang-update-ci-publish) ✗ VERSION=v0.1.1 BRANCH_NAME=master make -j2 publish
10:06:52 [ .. ] Pushing package xpkg.upbound.io/jastang/provider-argocd:v0.1.1
10:06:52 [ .. ] Skipping image publish for docker.io/provider-argocd:v0.1.1
Publish is deferred to xpkg machinery
10:06:52 [ OK ] Image publish skipped for docker.io/provider-argocd:v0.1.1
xpkg pushed to xpkg.upbound.io/jastang/provider-argocd:v0.1.1
10:07:04 [ OK ] Pushed package xpkg.upbound.io/jastang/provider-argocd:v0.1.1
```

*Verify*
```
➜  provider-argocd git:(jastang-update-ci-publish) crane manifest xpkg.upbound.io/jastang/provider-argocd:v0.1.1 | jq .
{
  "schemaVersion": 2,
  "manifests": [
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "size": 1005,
      "digest": "sha256:0a99f80d95dbd4196b18482eef1f84b5407599cf276a75c03ea2b3c39cbda395",
      "platform": {
        "architecture": "amd64",
        "os": "linux"
      }
    },
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "size": 1005,
      "digest": "sha256:3b14ebdc36c23ccb892ab52feb7934f3016e55262f3a14d39a6a24655edc534c",
      "platform": {
        "architecture": "arm64",
        "os": "linux"
      }
    }
  ]
}
```

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by jastang** [comment]
TODO:

- [ ] confirm dockerhub org `crossplane-contrib` is ready
- [ ] configure the Actions secrets for Upbound before running CI

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ env:
   # a step 'if env.AWS_USR' != ""', so we copy these to succinctly test whether
   # credentials have been provided before trying to run steps that need them.
   DOCKER_USR: ${{ secrets.DOCKER_USR }}
```
Comment (hasheddan) [inline .github/workflows/ci.yml:20]: Applying this across the board and also to `DOCKER_PSW` should work here 👍🏻 
```suggestion
  CONTRIB_DOCKER_USR: ${{ secrets.CONTRIB_DOCKER_USR }}
```

Comment (hasheddan) [reply .github/workflows/ci.yml:20]: Also will need to update https://github.com/crossplane-contrib/provider-argocd/blob/8fea55815aadd09c2c3f0bdbb2eaf323796608eb/Makefile#L58

**Review by hasheddan — COMMENTED** [review]
@jastang I've updated all necessary secrets here :)

See [this PR](https://github.com/crossplane-contrib/provider-aws/pull/1514) for example of remaining changes.

**Review by hasheddan — APPROVED** [review]
This looks awesome! Thanks @jastang!

**Pull request merged**